hera_qm/xrfi.py

```diff
@@ -1381,6 +1380,287 @@ def chi_sq_pipe(uv, alg='zscore_full_array', modified=False, sig_init=6.0,
+                    # if this is the first chunk, set uvf (metrics) equal to metrics chunk.
+                    if loadnum == 0:
+                        uvf = uvft
+                    # otherwise, combine metrics chunk with uvf aggragate.
+                    else:
+                        uvf.combine_metrics(uvft, method=wf_method, run_check=run_check,
+                                           check_extra=check_extra, run_check_acceptability=run_check_acceptability)
+            if run_filter:
+                # now that we have a uvf that includes the combined metric of all the baselines, we can
+                # run one last round of xrfi_pipe with flagging enabled, centering the metric enabled, and resetting
+                # the weights enabled to perform these final steps which are run on the full collapsed metric.
+                # note that we pass uvf as an arg instead of uv. xrfi_pipe has been modified so that if a uvflag
+                # is passed in place of uvdata or uvcal, it skips the metric calculation /waterfalling
+                # steps and goes straight to steps performed on combined metrics, i.e.
+                # flagging, normalizing, and weights reseting.
+                uvf, uvf_f = xrfi_pipe(uvf, alg=alg, Kt=kt_size, Kf=kf_size, xants=xants, skip_flags=False,
+                                         cal_mode=cal_mode, sig_init=sig_init, sig_adj=sig_adj, wf_method=wf_method,
+                                         center_metric=True, reset_weights=True,
+                                         label=label, run_check=run_check, check_extra=check_extra,
+                                         run_check_acceptability=run_check_acceptability)
+        # the following code is for when uv is a UVCal object.
+        elif issubclass(uv.__class__, UVCal):
+            # if uvf_apriori is not provided and we wish to derive it from uv
+            # do so here.
+            if uvf_apriori is None:
+                if calculate_uvf_apriori:
+                     uvf_apriori = UVFlag(uv, mode='flag', copy_flags=True, label='A priori flags.')
+                     uvf_apriori.to_waterfall(method='and', keep_pol=False, run_check=run_check,
+                                             check_extra=check_extra,
+                                             run_check_acceptability=run_check_acceptability)
+            # if uvf_apriori is not None and we wish to apply it to uv, do so here.
+            elif apply_uvf_apriori:
+                flag_apply(uvf_apriori, uv, keep_existing=True, run_check=run_check,
+                           check_extra=check_extra,
+                           run_check_acceptability=run_check_acceptability)
+            if run_filter:
+                # if run_filter is true, perform chi_sq_pipe or xrfi_pipe
+                if alg in ['zscore_full_array']:
+                    uvf, uvf_f = chi_sq_pipe(uv, alg=alg, modified=True, sig_init=sig_init, sig_adj=sig_adj,
```

Comment:
I can create an issue. Not sure what is meant by before and after this PR since from what I could tell, the previous versions of `xrfi_run` only computed modified overall z-scores.